src/serde_impls/deserializer.rs

```diff
@@ -193,1017 +193,1017 @@ macro_rules! delegate_except_bitseq {
+                .visit_seq(de::value::SeqDeserializer::new(values.into_iter().map(|(_, v)| v))),
+            Composite::Unnamed(values) => {
+                visitor.visit_seq(de::value::SeqDeserializer::new(values.into_iter()))
+            }
+        }
+    }
+
+    fn deserialize_tuple<V>(self, len: usize, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        match self {
+            // A sequence of named values? just ignores the names:
+            Composite::Named(values) => {
+                if values.len() != len {
+                    return Err(DeserializerError::from_string(format!(
+                        "Cannot deserialize composite of length {} into tuple of length {}",
+                        values.len(),
+                        len
+                    )));
+                }
+                visitor
+                    .visit_seq(de::value::SeqDeserializer::new(values.into_iter().map(|(_, v)| v)))
+            }
+            // A sequence of unnamed values is ideal:
+            Composite::Unnamed(values) => {
+                if values.len() != len {
+                    return Err(DeserializerError::from_string(format!(
+                        "Cannot deserialize composite of length {} into tuple of length {}",
+                        values.len(),
+                        len
+                    )));
+                }
+                visitor.visit_seq(de::value::SeqDeserializer::new(values.into_iter()))
+            }
+        }
+    }
+
+    fn deserialize_tuple_struct<V>(
+        self,
+        _name: &'static str,
+        len: usize,
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        self.deserialize_tuple(len, visitor)
+    }
+
+    fn deserialize_unit<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        // 0 length composite types can be treated as the unit type:
+        if self.is_empty() {
+            visitor.visit_unit()
+        } else {
+            Err(DeserializerError::from_str(
+                "Cannot deserialize non-empty Composite into a unit value",
+            ))
+        }
+    }
+
+    fn deserialize_unit_struct<V>(
+        self,
+        _name: &'static str,
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        self.deserialize_unit(visitor)
+    }
+
+    fn deserialize_newtype_struct<V>(
+        self,
+        _name: &'static str,
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        visitor.visit_seq(de::value::SeqDeserializer::new(Some(self).into_iter()))
+    }
+
+    fn deserialize_byte_buf<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        let mut bytes: Vec<u8> = Vec::new();
+        for v in self.into_values() {
```

Comment:
you could also use `self.into_values().map(....).collect::<Result<u8>, Self::Error>>` instead of manually constructing the vec....

Reply:
I went with that in the end, thanks!